### ISSUE TITLE
Error in using `core.apply_exclusion_zone` : non-precise type array

### ISSUE THREAD

**Issue by Ninimama** [TDAmeritrade/stumpy#514]
I used `core.core.apply_exclusion_zone` on the array `P=mp[:, 0]`, where `mp` is output of `stumpy.stump`. I got the following error:

```

~\AppData\Local\Temp/ipykernel_25128/138366328.py in stumpy_top_k_discords(T, m, k)
     37         k_discord_nn_idx[i] = mp[mp_discord_idx,1]
     38 
---> 39         core.apply_exclusion_zone(P, k_discord_idx[i], excl_zone, val=np.NINF)
     40 
     41     out = np.empty((k, 3), dtype = object)

~\anaconda3\lib\site-packages\numba\core\dispatcher.py in _compile_for_args(self, *args, **kws)
    413                 e.patch_message(msg)
    414 
--> 415             error_rewrite(e, 'typing')
    416         except errors.UnsupportedError as e:
    417             # Something unsupported is present in the user code, add help info

~\anaconda3\lib\site-packages\numba\core\dispatcher.py in error_rewrite(e, issue_type)
    356                 raise e
    357             else:
--> 358                 reraise(type(e), e, None)
    359 
    360         argtypes = []

~\anaconda3\lib\site-packages\numba\core\utils.py in reraise(tp, value, tb)
     78         value = tp()
     79     if value.__traceback__ is not tb:
---> 80         raise value.with_traceback(tb)
     81     raise value
     82 

TypingError: Failed in nopython mode pipeline (step: nopython frontend)
non-precise type array(pyobject, 1d, A)
During: typing of argument at C:\Users\nimas\anaconda3\lib\site-packages\stumpy\core.py (1449)

File "C:\Users\nimas\anaconda3\lib\site-packages\stumpy\core.py", line 1449:
def apply_exclusion_zone(a, idx, excl_zone, val=np.inf):
    <source elided>
    """
    zone_start = max(0, idx - excl_zone)
    ^


```
After some investigation, I realized that it can be resolved by just reconstructing `P` as follows: `P = np.array([x for x in P])`. 

However, this extra step does not change the type or shape of the array. So, I don't know the reason for such behavior.

**Comment by seanlaw** [comment]
@Ninimama Can you please confirm that you've already pulled the latest version from `main` into your branch?

**Comment by seanlaw** [comment]
I think the issue is that ` mp[:,0]` still have dtype `object`. So, you should try ` P = mp[:,0].astype(np.float64)` and then I think the apply_exclusion_zone will work

**Comment by seanlaw** [comment]
@Ninimama Would you mind pulling the latest version and trying again?

**Comment by Ninimama** [comment]
I thought `dtype` is reflected in `type()`. So, I just checked `type()` and thought things were good. 

> @Ninimama Would you mind pulling the latest version and trying again?

I pulled the latest version (and re-setup STUMPY and merge it to my branch). It is working when I change its dtype as you suggested `mp[:, 0].astype(np.float64)`. 

FYI: I am still getting error if I want to just use `mp[:,0]` with `dtype=object` and the **error message** is good!!!
